components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Runtime/Extensions/OpcConfigEx.cs

```diff
@@ -40,38 +41,58 @@ public static class OpcConfigEx {
 
             applicationConfiguration.CertificateValidator.CertificateValidation += handler;
 
-            X509Certificate2 certificate = null;
+            var configuredSubject = applicationConfiguration.SecurityConfiguration
+                .ApplicationCertificate.SubjectName;
+            applicationConfiguration.SecurityConfiguration.ApplicationCertificate.SubjectName = 
+                applicationConfiguration.ApplicationName;
+            applicationConfiguration.CertificateValidator
+                .Update(applicationConfiguration.SecurityConfiguration).ConfigureAwait(false);
 
-            // use existing certificate, if it is there
-            certificate = applicationConfiguration.SecurityConfiguration.ApplicationCertificate.Find(true).Result;
+            // use existing certificate, if present
+            var certificate = applicationConfiguration.SecurityConfiguration
+                .ApplicationCertificate.Certificate;
```

Comment:
>ApplicationCertificate.Certificate; [](start = 17, length = 35)

is it ok to not load the private key here?

Reply:
I think it is okay since we just validate if the application certificate exists in the store